actions/workflows/test_range.yaml

```diff
@@ -48,9 +60,62 @@ tasks:
       transport: "{{ ctx().transport }}"
       filter: "^{{ ctx().range_name }}$"
     next:
-      - when: "{{ succeeded() and (result().result.ranges.range[0].utilizationPercentage == 100) }}"
+      - when: "{{ succeeded() and (result().result.totalResults != 1) }}"
+        publish:
+          - err_message: "Range does not exist"
+        do:
+          - fail
+      - when: "{{ succeeded() and (result().result.totalResults == 1) and (result().result.ranges.range[0].utilizationPercentage == 100) }}"
+        publish:
+          - err_message: "Range is full"
+        do:
+          - fail
+      - when: "{{ succeeded() and (result().result.totalResults == 1) and (result().result.ranges.range[0].utilizationPercentage < 100) }}"
+        publish:
+          - starting_address: "{{ result().result.ranges.range[0].from }}"
+        do:
+          - reservation_dispatch
+
+  reservation_dispatch:
+    action: core.noop
+    next:
+      - when: "{{ succeeded() and not ctx().num_reserved_addresses }}"
+        do:
+          - noop
+      - when: "{{ succeeded() and ctx().range_name in ctx().excluded_ranges }}"
+        do:  
+          - noop
+      - when: "{{ succeeded() and (ctx().range_name not in ctx().excluded_ranges) and ctx().num_reserved_addresses }}"
+        publish:
+          - reserved_address_max_value: "{{ ctx().starting_address.split('.', 3)[3] | int + ctx().num_reserved_addresses }}"
+        do:  
+          - get_next_free_address
+  
+  get_next_free_address:
+    action: menandmice.get_next_free_address
+    input:
+      server: "{{ ctx().server }}"
+      session: "{{ ctx().session }}"
+      transport: "{{ ctx().transport }}"
+      range_ref: "{{ ctx().range_name }}"
+      start_address: "{{ ctx().starting_address }}"
+      temporary_claim_time: "{{ ctx().claim_time }}"
+    next:
+      - when: "{{ succeeded() }}"
+        publish:
+          - next_free_octet: "{{ result().result.split('.', 3)[3] | int }}"
+        do:
+          - check_addresses
+
+  check_addresses:
+    action: core.noop
+    next:
+      - when: "{{ succeeded() and ctx().next_free_octet <= ctx().reserved_address_max_value }}"
+        publish:
+          - err_message: "addresses are not claimed"
```

Comment:
we should add into this error message about what we are expecting, thinking about receiving an email with this as the error:

"addresses are not claimed"

Does that give you enough information about what is wrong and what the system is expecting to see, or even where to look about how to fix the error?

A better error message would be something like:
```
Error claiming an IP address in Men&Mice within the range 1.2.3.0/24. When trying to claim a new IP address in this range, we received an address of 1.2.3.1. This is an error. It is expected that the first 15 addresses [1.2.3.1 - 1.2.3.15] are either marked as claimed or assigned a DNS record so that new systems do not step on existing network equipment, such as the default gateway. To fix this error, please go into Men&Mice and claim the first 15 addresses within the range 1.2.3.0/24.